### PR TITLE
lib-user: Add group create modal

### PR DESCRIPTION
**Pull request by mcbouslog**
## Package
- lib-react-components
- lib-user

## Describe your changes
- add create group button
- add group modal
- add group create form

## How to Review
_Helpful explanations that will make your reviewer happy:_
- What user actions should my reviewer step through to review this PR?
  - create a new staging group from local lib-user (`yarn dev`): https://local.zooniverse.org:8080/?users=[insert your staging user login here]/groups
  - create a new production group from local app-root (`yarn start`): https://local.zooniverse.org:3000/users/[insert your production user login here]/groups
- Which storybook stories should be reviewed?
  - create button - http://localhost:6008/?path=/story/components-mygroups-createbutton--default
  - group modal - http://localhost:6008/?path=/story/components-mygroups-groupmodal--create-group
  - group create form - http://localhost:6008/?path=/story/components-mygroups-groupform--default
- Are there plans for follow up PR’s to further fix this bug or develop this feature?
  - these changes don't include styling changes per Figma designs, PR was getting quite large, holding off on design edits to include with group manage modal

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General
- [ ] Tests are passing locally and on Github
- [ ] Documentation is up to date and changelog has been updated if appropriate
- [ ] You can `yarn panic && yarn bootstrap` or `docker-compose up --build` and FEM works as expected
- [ ] FEM works in all major desktop browsers: Firefox, Chrome, Edge, Safari (Use Browserstack account as needed)
- [ ] FEM works in a mobile browser

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)
- [ ] Can sign-in and sign-out
- [ ] The component is accessible
  - Can be used with a screen reader [BBC guide to VoiceOver](https://bbc.github.io/accessibility-news-and-you/assistive-technology/testing-steps/voiceover-mac.html)
  - Can be used from the keyboard [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - It is passing accessibility checks in the storybook

## New Feature
- [ ] The PR creator has listed user actions to use when testing the new feature
- [ ] Unit tests are included for the new feature
- [ ] A storybook story has been created or updated
  - Example of SlideTutorial [component](https://github.com/zooniverse/front-end-monorepo/blob/master/packages/lib-classifier/src/components/Classifier/components/SlideTutorial/SlideTutorial.js) with docgen comments, and its [story](https://zooniverse.github.io/front-end-monorepo/@zooniverse/classifier/index.html?path=/docs/other-slidetutorial--default)

### CODE REVIEW

File: packages/lib-user/src/components/MyGroups/components/GroupForm/GroupForm.js [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+  return (
+    <Box
+      width={{ min: '600px' }}
+    >
+      <Form
+        onChange={(nextValue, { touched }) => {
+          if (nextValue.visibility !== value.visibility) {
+            const statsVisibility = nextValue.visibility === 'Private' ? 'private_agg_only' : 'public_agg_only'
+            nextValue.stats_visibility = statsVisibility
+          }
```
Comment (mcbouslog) [inline packages/lib-user/src/components/MyGroups/components/GroupForm/GroupForm.js:60]: This is setting the `stats_visibility` options per the Public or Private radio button group. The Public/Private field is not saved or part of a new user group, but used to simplify the `stats_visibility` options.

File: packages/lib-react-components/src/Modal/Modal.js [hunk 2]
```diff
@@ -6,6 +6,10 @@ import ModalBody from './components/ModalBody'
 import ModalHeading from './components/ModalHeading'
 
 const Modal = forwardRef(function ({
+  bodyBackground = {
+    dark: 'dark-5',
+    light: 'neutral-6'
+  },
```
Comment (goplayoutside3) [inline packages/lib-react-components/src/Modal/Modal.js:12]: Suggestion to define this style object outside of the Modal component like:
```
const defaultBg = {
    dark: 'dark-5',
    light: 'neutral-6'
}
```

You could pass it to the parent Box where `background` is defined in this file too. Similar strategy to:
https://github.com/zooniverse/front-end-monorepo/blob/cf0fcfc2630707555294760c7af07fd58e76768c/packages/lib-react-components/src/AuthModal/components/LoginForm/LoginForm.js#L7-L10

File: packages/lib-react-components/src/Modal/Modal.js [hunk 3]
```diff
@@ -52,6 +57,10 @@ Modal.propTypes = {
     Determines whether the modal is visible or not.
   */
   active: PropTypes.bool,
+  /**
+    The background color for the modal body. It can be set to any CSS color value or color string value from the Zooniverse Grommet theme or an object setting the color for the light and dark theme.
+  */
```
Comment (goplayoutside3) [inline packages/lib-react-components/src/Modal/Modal.js:62]: Just noting relation to this Issue: https://github.com/zooniverse/front-end-monorepo/issues/1798

**Review by goplayoutside3 — APPROVED** [review]
Nice! I can create a new group when signed-in to app-root or lib-user 👍 

Only design note is to consider including `dark: white` or similar for the Create button:
Screenshot 2024-04-04 at 3 18 00 PM

**Pull request merged**